exporter/signalfxexporter/exporter.go

```diff
@@ -230,10 +230,12 @@ func (se *signalfxExporter) startLogs(_ context.Context, host component.Host) er
 func (se *signalfxExporter) createClient(host component.Host) (*http.Client, error) {
 	se.config.HTTPClientSettings.TLSSetting = se.config.IngestTLSSettings
 
-	if se.config.HTTPClientSettings.MaxIdleConns == nil {
+	if se.config.HTTPClientSettings.MaxIdleConns == nil && se.config.MaxConnections != 0 {
+		se.logger.Warn("You are using the deprecated `max_connections` option that will be removed soon; use `max_idle_conns` instead: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/signalfxexporter#advanced-configuration")
 		se.config.HTTPClientSettings.MaxIdleConns = &se.config.MaxConnections
 	}
-	if se.config.HTTPClientSettings.MaxIdleConnsPerHost == nil {
+	if se.config.HTTPClientSettings.MaxIdleConnsPerHost == nil && se.config.MaxConnections != 0 {
+		se.logger.Warn("You are using the deprecated `max_connections` option that will be removed soon; use `max_idle_conns_per_host` instead: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/signalfxexporter#advanced-configuration")
 		se.config.HTTPClientSettings.MaxIdleConnsPerHost = &se.config.MaxConnections
 	}
```

Comment:
It'll bring two messages to everyone who uses just `max_connections`, which is the most common use case. I think it should be

```golang
if  se.config.MaxConnections != 0 && (se.config. MaxIdleConns == nil || se.config.HTTPClientSettings.MaxIdleConnsPerHost == nil) {
	se.logger.Warn("You are using the deprecated `max_connections` option that will be removed soon; use `max_idle_conns` and/or `max_idle_conns_per_host` instead: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/signalfxexporter#advanced-configuration")
	if se.config.HTTPClientSettings.MaxIdleConns == nil {
		se.config.HTTPClientSettings.MaxIdleConns = &se.config.MaxConnections
	}
	if se.config.HTTPClientSettings. MaxIdleConnsPerHost == nil {
		se.config.HTTPClientSettings. MaxIdleConnsPerHost = &se.config.MaxConnections
	}
}
```

Reply:
Makes sense, applied.